ui/src/app/applications/components/applications-list/applications-filter.tsx

```diff
@@ -1,238 +1,164 @@
-                                            applications.forEach(app => projAppCount.set(app.spec.project, (projAppCount.get(app.spec.project) || 0) + 1));
-                                            return (
-                                                <TagsInput
-                                                    placeholder='default'
-                                                    autocomplete={projects.map(proj => proj.metadata.name)}
-                                                    tags={pref.projectsFilter}
-                                                    onChange={selected => onChange({...pref, projectsFilter: selected})}
-                                                />
-                                            );
-                                        }}
-                                    </DataLoader>
-                                </li>
-                            </ul>
-                            <div className='applications-list__filter-title'>Clusters</div>
-                            <ul>
-                                <li>
-                                    <TagsInput
-                                        placeholder='in-cluster (https://kubernetes.default.svc)'
-                                        autocomplete={Array.from(new Set(applications.map(app => this.getClusterDetail(app.spec.destination)).filter(item => !!item))).filter(
-                                            ns => pref.clustersFilter.indexOf(ns) === -1
-                                        )}
-                                        tags={pref.clustersFilter}
-                                        onChange={selected => onChange({...pref, clustersFilter: selected})}
-                                    />
-                                </li>
-                            </ul>
-                            <div className='applications-list__filter-title'>Namespaces</div>
-                            <ul>
-                                <li>
-                                    <TagsInput
-                                        placeholder='*-us-west-*'
-                                        autocomplete={Array.from(new Set(applications.map(app => app.spec.destination.namespace).filter(item => !!item))).filter(
-                                            ns => pref.namespacesFilter.indexOf(ns) === -1
-                                        )}
-                                        tags={pref.namespacesFilter}
-                                        onChange={selected => onChange({...pref, namespacesFilter: selected})}
-                                    />
-                                </li>
-                            </ul>
-                        </div>
-                    </div>
-                </div>
-            </div>
-        );
-    }
+const ProjectFilter = (props: AppFilterProps) => {
+    const [projects] = useData(() => services.projects.list('items.metadata.name'));
```

Comment:
Please render loading and error state for both clusters and projects filter. In production argo cd instances both API requests might take quite a lot of time.